### PR TITLE
Add Otto Matic 4.0.0

### PR DESCRIPTION
**Pull request by ivanholmes**
Based on #6891 #6897 #6899

### CODE REVIEW

**Comment by ivanholmes** [comment]
Ready to merge from my POV.

File: games-action/ottomatic/ottomatic-4.0.0.recipe [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+CHECKSUM_SHA256="6aff7ac6bd8b08e04e6927c59a2e5b77bfc49c814bbc95aef6efb6d92cf6b10a"
+SOURCE_DIR="OttoMatic-$portVersion"
+PATCHES="ottomatic-$portVersion.patchset"
+ADDITIONAL_FILES="ottomatic.rdef.in"
+srcGitRev2="b9ddab06cdc0b9e4a8085e50974213b6c9625436"
+SOURCE_URI_2="https://github.com/jorio/Pomme/archive/$srcGitRev2.tar.gz"
+CHECKSUM_SHA256_2="96be755cbbcc2f4a3677dd4388599bad4b5cc771a55d927bb8ebe646d7142b52"
+SOURCE_FILENAME_2="Pomme-$srcGitRev2.tar.gz"
+SOURCE_DIR_2="Pomme-$srcGitRev2"
+PATCHES_2="ottomatic-$portVersion-source2.patchset"
```
Comment (korli) [inline games-action/ottomatic/ottomatic-4.0.0.recipe:26]:
```suggestion
SOURCE_URI="https://github.com/jorio/OttoMatic/archive/refs/tags/$portVersion.tar.gz"
CHECKSUM_SHA256="6aff7ac6bd8b08e04e6927c59a2e5b77bfc49c814bbc95aef6efb6d92cf6b10a"
SOURCE_DIR="OttoMatic-$portVersion"
PATCHES="ottomatic-$portVersion.patchset"
srcGitRev2="b9ddab06cdc0b9e4a8085e50974213b6c9625436"
SOURCE_URI_2="https://github.com/jorio/Pomme/archive/$srcGitRev2.tar.gz"
CHECKSUM_SHA256_2="96be755cbbcc2f4a3677dd4388599bad4b5cc771a55d927bb8ebe646d7142b52"
SOURCE_FILENAME_2="Pomme-$srcGitRev2.tar.gz"
SOURCE_DIR_2="Pomme-$srcGitRev2"
PATCHES_2="ottomatic-$portVersion-source2.patchset"
ADDITIONAL_FILES="ottomatic.rdef.in"
```

File: games-action/ottomatic/ottomatic-4.0.0.recipe [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	ottomatic$secondaryArchSuffix = $portVersion
+	app:OttoMatic = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libsdl2_2.0$secondaryArchSuffix
+	lib:libgl$secondaryArchSuffix
```
Comment (korli) [inline games-action/ottomatic/ottomatic-4.0.0.recipe:38]:
```suggestion
	lib:libgl$secondaryArchSuffix
	lib:libsdl2_2.0$secondaryArchSuffix
```

File: games-action/ottomatic/ottomatic-4.0.0.recipe [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libsdl2_2.0$secondaryArchSuffix
+	devel:libgl$secondaryArchSuffix
```
Comment (korli) [inline games-action/ottomatic/ottomatic-4.0.0.recipe:44]:
```suggestion
	devel:libgl$secondaryArchSuffix
	devel:libsdl2_2.0$secondaryArchSuffix
```

File: games-action/ottomatic/ottomatic-4.0.0.recipe [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
```
Comment (korli) [inline games-action/ottomatic/ottomatic-4.0.0.recipe:49]:
```suggestion
	cmd:gcc$secondaryArchSuffix
	cmd:make
```

File: games-action/ottomatic/ottomatic-4.0.0.recipe [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	"
+
+BUILD()
+{
+	# submodule
+	rm -rf extern/Pomme
+	ln -s $sourceDir2 extern/Pomme
+	# building
+	cmake -S . -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo
```
Comment (korli) [inline games-action/ottomatic/ottomatic-4.0.0.recipe:58]:
```suggestion
	cmake -S . -B build -DCMAKE_BUILD_TYPE=Release
```

**Pull request merged**